pkg/state/infra.go

```diff
@@ -143,5 +517,122 @@ func (i *infraClient) Shutdown() error {
 	for _, router := range i.routers {
 		router.Shutdown()
 	}
+
+	for _, broker := range i.brokers {
+		broker.Shutdown()
+	}
+	return nil
+}
+
+func (i *infraClient) DeleteEndpoint(endpoint *v1beta2.MessagingEndpoint) error {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	ctx := context.Background()
+	for _, internalPort := range endpoint.Status.InternalPorts {
+		if internalPort.Protocol == v1beta2.MessagingProtocolAMQP {
+			err := i.applyRouters(ctx, func(router *RouterState) error {
+				return router.DeleteListener(internalPort.Name)
+			})
+			if err != nil {
+				return err
+			}
+		} else {
+			// TODO
+			return fmt.Errorf("%s protocol not yet supported", internalPort.Protocol)
+		}
+		i.freePort(internalPort.Port)
+	}
```

Comment:
what is responsible for cleaning the entry from the e.endpoint map?

Reply:
Thats an oversight on my part. It should be in DeleteEndpoint (and same for DeleteAddress which is also missing the delete).